client/lib/wp/browser.js

```diff
@@ -14,12 +15,16 @@ import wpcomSupport from 'lib/wp/support';
 let wpcom;
 
 if ( config.isEnabled( 'oauth' ) ) {
-	const oauthToken = require( 'lib/oauth-token' ),
-		requestHandler = require( 'lib/wpcom-xhr-wrapper' );
+	const oauthToken = require( 'lib/oauth-token' );
+	const requestHandler = config.isEnabled( 'sync-handler' )
```

Comment:
The thing here is you also need to decide what request handler will be wrapped by sync-handler. It's a double nested condition. And the wpcom instantiation is different accordingly. 

```jsx
if ( config.isEnabled( 'oauth' ) ) {
  // ...
  if( config.isEnabled( 'sync-handler' ) ) {
    const req... = ...
  } else {
    const....
  }
} else {
    // ...
  if( config.isEnabled( 'sync-handler' ) ) {
    const req... = ...
  } else {
    const....
  }
}
```
I can add